processing/src/main/java/org/carbondata/processing/store/writer/CarbonFactDataWriterImplForIntIndexAndAggBlock.java

```diff
@@ -106,12 +111,36 @@ public NodeHolder buildDataNodeHolder(IndexStorage<int[]>[] keyStorageArray, byt
 
       }
       totalKeySize += keyLengths[i];
-      if (isComplexType[i] || isDictionaryColumn[i]) {
-        allMinValue[i] = keyStorageArray[i].getMin();
-        allMaxValue[i] = keyStorageArray[i].getMax();
+      if(isUseInvertedIndex[i]) {
+        if (isComplexType[i] || isDictionaryColumn[i]) {
+          allMinValue[i] = keyStorageArray[i].getMin();
+          allMaxValue[i] = keyStorageArray[i].getMax();
+        } else {
+          allMinValue[i] = updateMinMaxForNoDictionary(keyStorageArray[i].getMin());
+          allMaxValue[i] = updateMinMaxForNoDictionary(keyStorageArray[i].getMax());
+        }
       } else {
-        allMinValue[i] = updateMinMaxForNoDictionary(keyStorageArray[i].getMin());
-        allMaxValue[i] = updateMinMaxForNoDictionary(keyStorageArray[i].getMax());
+        byte[][] minmaxArray = keyStorageArray[i].getKeyBlock();
```

Comment:
Move the code of calculating min max to the `BlockIndexerStorageForNoInvertedIndex`. It already has method to get min and max so better calculate there and make it available to `getMin` and `getMax` methods.